yt/frontends/boxlib/fields.py

```diff
@@ -523,15 +525,21 @@ def setup_fluid_fields(self):
 
 
 def _nice_species_name(field):
-    spec_match = spec_finder.search(field)
-    nice_name = "".join(spec_match.groups())
-    # if the species field is a descriptive name, then the match
-    # on the integer will be blank
-    # modify the tex string in this case to remove spurious tex spacing
-    lab = r"X\left(^{%s}%s\right)"
-    if spec_match.groups()[-1] == "":
-        lab = r"X\left(%s%s\right)"
-    tex_label = lab % spec_match.groups()[::-1]
+    try:
+        spec_match = spec_finder.search(field)
+        nice_name = "".join(spec_match.groups())
+        # if the species field is a descriptive name, then the match
+        # on the integer will be blank
+        # modify the tex string in this case to remove spurious tex spacing
+        lab = r"X\left(^{%s}%s\right)"
+        if spec_match.groups()[-1] == "":
+            lab = r"X\left(%s%s\right)"
+        tex_label = lab % spec_match.groups()[::-1]
+
+    except:
```

Comment:
Bare except statements should be avoided, please only catch specific exceptions. The code in the `try` branch should also be limited to the part that's expected to sometimes raise exceptions.